src/blueprints/character/character_routes.py

```diff
@@ -0,0 +1,88 @@
+from flask import render_template, request, redirect, url_for, session
+import logging
+from services import character_service
+from utils.form_handlers import CharacterHandlers
+from utils.csrf_utils import get_csrf_token
+from . import character_blueprint
+
+
+@character_blueprint.route('/characters/list', methods=['GET'])
+def my_characters_route():
+    error = None
+    user_id = session.get('user_id')
+
+    try:
+        character_list = character_service.get_characters_by_user_id(user_id)
+    except Exception as e:
+        error = "Character list fetch failed."
+        logging.error(f"Error occurred: {str(e)}")
+
+    csrf_token = get_csrf_token()
+    return render_template('my_characters.html', character_list=character_list, error=error, csrf_token=csrf_token)
+
+
+@character_blueprint.route('/characters/create', methods=['GET'])
+def create_character_route_get():
+    error = None
+
+    csrf_token = get_csrf_token()
+    return render_template('create_character.html', error=error, csrf_token=csrf_token, character=None)
+
+
+@character_blueprint.route('/characters/create', methods=['POST'])
+def create_character_route_post():
+    error = None
+
+    name, health, armor_class = CharacterHandlers.get_character_form_data()
+    user_id = session.get('user_id')
+
+    error = CharacterHandlers.character_form_check_input(name, health, armor_class, error)
+
+    if not error:
+        try:
+            character_service.create_character(name, health, armor_class, user_id)
+            return redirect(url_for('my_characters_route'))
+        except Exception as e:
+            error = "Character creation failed."
+            logging.error(f"Error occurred: {str(e)}")
+    
+
+@character_blueprint.route('/characters/modify/<int:character_id>', methods=['GET'])
+def modify_character_get(character_id):
+    error = None
+    csrf_token = get_csrf_token()
+
+    character = character_service.get_character_by_character_id(character_id)
+
+    if character is None:
+        error = "Character not found"
+        return redirect('my_characters_route', error=error, csrf_token=csrf_token)
+    else:
+        return render_template('modify_character.html', error=error, character_id=character_id, character=character, csrf_token=csrf_token)
```

Comment:
A suggestion that comes down to personal prefence. Instead of having an else block, it can just be omitted.

Personally I feel that it often leads to more readable code. Of course all paths before it must return for  it to work.

```suggestion
    if character is None:
        error = "Character not found"
        return redirect('my_characters_route', error=error, csrf_token=csrf_token)
       
    return render_template('modify_character.html', error=error, character_id=character_id, character=character, csrf_token=csrf_token)
```